mlflow/gateway/client.py

```diff
@@ -344,3 +345,77 @@ def query(self, route: str, data: Dict[str, Any]):
                 raise MlflowException(message=timeout_message, error_code=BAD_REQUEST)
             else:
                 raise e
+
+    @experimental
+    def set_limits(self, route: str, limits: List[Dict[str, Any]] = []) -> LimitsConfig:
+        """
+        Set limits on an existing route in the Gateway.
+
+        .. warning::
+
+            This API is **only available** when running within Databricks.
+
+        :param route_name: The name of the route to set limits on.
+        :param limits: Limits to set on the route.
+
+        :raises mlflow.MlflowException: If the function is not running within Databricks.
+
+        Example usage from within Databricks:
+
+        .. code-block:: python
+
+            from mlflow.gateway import set_gateway_uri, set_limits
+
+            set_gateway_uri(gateway_uri="databricks")
+
+            set_limits("my-new-route", [{"key": "user", "renewal_period": "minute", "calls": 50}])
+
+        """
+
+        if not self._is_databricks_host():
+            raise MlflowException(
+                "The set_limits API is only available when running within Databricks.",
+                error_code=BAD_REQUEST,
+            )
+        payload = {
+            "route": route,
+            "limits": limits,
+        }
+
+        response = self._call_endpoint(
+            "POST", MLFLOW_GATEWAY_LIMITS_BASE, json.dumps(payload)
+        ).json()
+        return LimitsConfig(**response)
+
+    @experimental
+    def get_limits(self, route: str) -> LimitsConfig:
+        """
+        Get limits of an existing route in the Gateway.
+
+        .. warning::
+
+            This API is **only available** when running within Databricks.
+
+        :param route: The name of the route to get limits of.
+
+        :raises mlflow.MlflowException: If the function is not running within Databricks.
+
+        Example usage from within Databricks:
+
+        .. code-block:: python
+
+            from mlflow.gateway import set_gateway_uri, get_limits
+
+            set_gateway_uri(gateway_uri="databricks")
+            get_limits("my-new-route")
+
+        """
+
+        if not self._is_databricks_host():
```

Comment:
Can we add routes for limits in OSS and make them respond with an error like "get_limits is unsupported"?

Reply:
Sorry, do you mean we need to check if a route is in OSS and if yes respond with an error?

Reply:
Wondering if users set_gateway_uri("databricks"), can they have OSS routes (sorry I dont have much context here)?

Reply:
If there is no plan to support this in OSS, yeah the current approach seems reasonable.

Reply:
Not a blocker. Just an idea to avoid limiting the client compatibility.

Reply:
Hope someone contributes limits in OSS (https://github.com/mlflow/mlflow/issues/9456)

Reply:
@zhe-db I prefer @harupy 's suggestion here. This way, once we add limits support to Gateway, old clients can interact with them.

Reply:
sg! will address it!